### PR TITLE
Adding generic and new entity repository

### PR DESCRIPTION
**Pull request by ArtyomBanchikov**

### CODE REVIEW

File: Pets.DAL/Repositories/PetRepository.cs [hunk 1]
```diff
@@ -2,10 +2,10 @@
 using Pets.DAL.Contexts;
 using Pets.DAL.Entities;
 using Pets.DAL.Interfaces;
-
+    
 namespace Pets.DAL.Repositories
 {
-    public class PetRepository: IPetRepository
+    public class PetRepository : IPetRepository
```
Comment (stas1408123) [inline Pets.DAL/Repositories/PetRepository.cs]: Inherit from GenericRepository<PetEntyty> and IPetRepository

Comment (stas1408123) [reply Pets.DAL/Repositories/PetRepository.cs]: And you don't need to implement methods

**Review by stas1408123 — CHANGES_REQUESTED** [review]

File: Pets.DAL/Interfaces/IFoodRepository.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using Pets.DAL.Entities;
+
+namespace Pets.DAL.Interfaces
+{
+    public interface IFoodRepository: IGenericRepository<FoodEntity>
```
Comment (hromau) [inline Pets.DAL/Interfaces/IFoodRepository.cs]: why it is so empty?

**Review by hromau — CHANGES_REQUESTED** [review]

File: Pets.DAL/Interfaces/IFoodRepository.cs (hunk 2 above)
Comment (ArtyomBanchikov) [reply Pets.DAL/Interfaces/IFoodRepository.cs]: There is all realisation in base interface

Comment (hromau) [reply Pets.DAL/Interfaces/IFoodRepository.cs]: it isn't reason make an empty interface or class

**Review by hromau — CHANGES_REQUESTED** [review]

File: Pets.DAL/Interfaces/IFoodRepository.cs (hunk 2 above)
Comment (ArtyomBanchikov) [reply Pets.DAL/Interfaces/IFoodRepository.cs]: I use this empty interface for DI

**Review by stas1408123 — APPROVED** [review]

**Review by hromau — APPROVED** [review]

**Pull request merged**